tests/kernel/usleep/src/main.c

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019 Intel Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <zephyr.h>
+
+#define SLEEP_US 	50
+#define LOOPS 		200
+
+#define LOWER_BOUND_MS	((SLEEP_US * LOOPS) / 1000)
+#define UPPER_BOUND_MS	(LOWER_BOUND_MS * 2)
```

Comment:
Seems like a factor of two is more than you need.  The lower bound is safe, but this one could be tightened quite a bit.  By my read the test below should be accurate to +/- 2 ticks.

Reply:
I agree, but I can't get the test past QEMU reliably without the wider margin.  Real hardware, no problem.  Emulation.. not so much.

Reply:
Qemu is a problem regardless.  This doesn't look to me like it's particularly "more" susceptible to the host interactions than other tests.    But sure.  We can always tighten things later.